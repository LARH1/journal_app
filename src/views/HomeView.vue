<template>
<div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
    <button class="btn btn-success" @click="Cambiar">success</button>
    <button class="btn btn-primary" @click="Cambiar">primary</button>
    <button class="btn btn-danger" @click="Cambiar">danger</button>
</div>
</template>

<script>
export default
{
    methods:
    {
        Cambiar()
        {
            this.$router.push(
            {
                name: "empty-entry"
            });
        },
    },
};
</script>
