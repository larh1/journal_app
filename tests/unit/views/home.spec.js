import {shallowMount} from "@vue/test-utils";
import HomeView from "@/views/HomeView"

describe("Home test",() =>
{
    test("Debe coincidir con el spanshot",() =>
    {
        const wrap1 = shallowMount(HomeView)
        expect(wrap1.html()).toMatchSnapshot();
    });

    test("Al hacer clic debe cambiar a 'empty-entry'",() =>
    {
        // Crear mock del router
        const mockRouter = {
            push: jest.fn(),
        };
        // Agregarlo al component
        const wrap = shallowMount(HomeView,{
            global:
            {
                mocks:
                {
                    $router: mockRouter // El router del Home
                }
            }
        });
        // Llamar evento click del boton
        wrap.find("button").trigger("click");
        // Se espera que el click haya sido llamado
        expect(mockRouter.push).toHaveBeenCalled();
        // Se espera que el click haya sido llamado con la data
        expect(mockRouter.push).toHaveBeenCalledWith({name:"empty-entry"});
    });
})