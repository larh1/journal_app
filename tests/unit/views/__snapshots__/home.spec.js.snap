// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test Debe coincidir con el spanshot 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-success">success</button><button class="btn btn-primary">primary</button><button class="btn btn-danger">danger</button></div>`;
